Services/Diagnostics/DiagnosticsLogger.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Http;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Runtime;
+
+namespace Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Diagnostics
+{
+    public interface IDiagnosticsLogger
+    {
+        Task<IHttpResponse> LogServiceStartAsync(string message);
+
+        Task<IHttpResponse> LogServiceHeartbeatAsync();
+
+        Task<IHttpResponse> LogServiceErrorAsync(
+            string message,
+            [CallerMemberName] string callerName = "",
+            [CallerFilePath] string filePath = "",
+            [CallerLineNumber] int lineNumber = 0);
+
+        Task<IHttpResponse> LogServiceErrorAsync(
+            string message,
+            object data,
+            [CallerMemberName] string callerName = "",
+            [CallerFilePath] string filePath = "",
+            [CallerLineNumber] int lineNumber = 0);
+
+        Task<IHttpResponse> LogServiceExceptionAsync(
```

Comment:
what's the difference between and Error and and Exception?

Reply:
I think we could merge the 2, or at least use the same name. When logging an exception we might want to attache data, e.g. exceptionMessage could be part of data

Reply:
renamed it :) done